src/selectors/entities/channels.js

```diff
@@ -754,3 +823,172 @@ export const filterPostIds = (condition) => {
         }
     );
 };
+
+const getAllActiveChannels = createSelector(
+    getPublicChannels,
+    getPrivateChannels,
+    getDirectChannels,
+    (
+        publicChannels,
+        privateChannels,
+        directChannels,
+    ) => {
+        const allChannels = [
+            ...publicChannels,
+            ...privateChannels,
+            ...directChannels,
+        ];
+
+        return allChannels;
+    }
+);
+
+export const getAllChannelIds = createIdsSelector(
+    getAllActiveChannels,
+    getCurrentUser,
+    getMyChannelMemberships,
+    getLastPostPerChannel,
+    (state, lastUnreadChannel, unreadsAtTop, favoritesAtTop, sorting = 'alpha') => sorting,
+    mapAndSortChannelIds,
+);
+
+export const getAllSortedChannelIds = createIdsSelector(
+    getUnreadChannelIds,
+    getFavoritesPreferences,
+    getAllChannelIds,
+    (state, lastUnreadChannel, unreadsAtTop = true) => unreadsAtTop,
+    (state, lastUnreadChannel, unreadsAtTop, favoritesAtTop = true) => favoritesAtTop,
+    filterChannels,
+);
+
+export const getOrderedChannelIds = (state, lastUnreadChannel, grouping, sorting, unreadsAtTop, favoritesAtTop) => {
+    const channels = [];
+
+    if (grouping === 'by_type') {
+        channels.push({
+            type: 'public',
+            name: 'PUBLIC CHANNELS',
```

Comment:
It doesn't look like this `name` field is being used anywhere, so it can probably be removed. If it is being used, can you make sure that wherever is using it is localizing it properly?

Reply:
It's used in the webapp and it's being localized